__tests__/distributors/adoptium-installer.test.ts

```diff
@@ -0,0 +1,164 @@
+import { HttpClient } from '@actions/http-client';
+
+import {
+  AdoptiumDistribution,
+  AdoptiumImplementation
+} from '../../src/distributions/adoptium/installer';
+import { JavaInstallerOptions } from '../../src/distributions/base-models';
+
+let manifestData = require('../data/adoptium.json') as [];
+
+describe('getAvailableVersions', () => {
+  let spyHttpClient: jest.SpyInstance;
+
+  beforeEach(() => {
+    spyHttpClient = jest.spyOn(HttpClient.prototype, 'getJson');
+    spyHttpClient.mockReturnValue({
+      statusCode: 200,
+      headers: {},
+      result: []
+    });
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+    jest.clearAllMocks();
+    jest.restoreAllMocks();
+  });
+
+  it.each([
+    [
+      { version: '16', architecture: 'x64', packageType: 'jdk', checkLatest: false },
+      AdoptiumImplementation.Hotspot,
```

Comment:
s/Adoptium/Temurin/g in this file. Also, consider dropping the .Hotspot part as it is not needed for Temurin.